### PR TITLE
[SER-486] Test string serialization

### PR DESCRIPTION
**Pull request by TheButlah**

### CODE REVIEW

File: demos/unity_states/cs/.gitignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /src/bin/
 /src/obj
-/src/generated/
+
+/generated/
```
Comment (tomthecarrot) [inline demos/unity_states/cs/.gitignore]: Generated files from the `build_mac_host` script are placed in `/src/generated`. What is the purpose of this change?

**Review by tomthecarrot — CHANGES_REQUESTED** [review]
tested locally and works. one q:

File: demos/unity_states/cs/.gitignore (hunk 1 above)
Comment (TheButlah) [reply demos/unity_states/cs/.gitignore]: on my machine they were placed in the changed folder. Which is the correct location?

Comment (tomthecarrot) [reply demos/unity_states/cs/.gitignore]: `/src/generated` is the correct location as executed in the build script. Can you post the output of `tree generated/`?

Comment (TheButlah) [reply demos/unity_states/cs/.gitignore]: resolved. Not sure why I thought it needed changing.

**Review by tomthecarrot — APPROVED** [review]
